### PR TITLE
Kafka connect helm chart

### PR DESCRIPTION
**Pull request by yash-saxena-d11**
### Summary
Kafka connect helm chart

### Full changelog
* Add helm chart for kafka connect

### CODE REVIEW

File: charts/kafka-connect/templates/NOTES.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+CHART NAME: {{ .Chart.Name }}
+CHART VERSION: {{ .Chart.Version }}
+APP VERSION: {{ .Chart.AppVersion }}
+
+** Please be patient while the chart is being deployed **
+
+Check the cluster status by running:
+
+   kubectl exec -it --namespace {{ .Release.Namespace }} $(kubectl get pods --namespace {{ .Release.Namespace }} -l app.kubernetes.io/name={{ include "common.names.name" . }},app.kubernetes.io/instance={{ .Release.Name }} -o jsonpath='{.items[0].metadata.name}') -- asadm
```
Comment (akshaypatidar1999) [inline charts/kafka-connect/templates/NOTES.txt:9]: Update notes according to kafka connect

**Comment by akshaypatidar1999** [comment]
Add some examples in `ci` directory

File: charts/kafka-connect/README.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+| `nameOverride`                                             | String to partially override common.names.fullname                            | `""`                                                               |
+| `fullnameOverride`                                         | String to fully override common.names.fullname                                | `""`                                                               |
+| `commonLabels`                                             | Labels to add to all deployed objects (sub-charts are not considered)         | `{}`                                                               |
+| `commonAnnotations`                                        | Annotations to add to all deployed objects                                    | `{}`                                                               |
+| `replicaCount`                                             | to decide the number of kafka connect pod replicas                            | `1`                                                                |
```
Comment (akshaypatidar1999) [inline charts/kafka-connect/README.md:50]: Description should start with capital letter

File: charts/kafka-connect/values.yaml [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+  "offset.storage.replication.factor": "3"
+  ##  @param configurationOverrides.status.storage.replication.factor status storage replication factor
+  "status.storage.replication.factor": "3"
+
+## @param heapOptions Kafka Connect JVM Heap Option
+heapOptions: "-Xms512M -Xmx512M"
+
+## Additional env variables
+## @param customEnv Additional env variables
+customEnv: {}
```
Comment (akshaypatidar1999) [inline charts/kafka-connect/values.yaml:151]: Rename this to `extraEnvVars` to be consistent with aerospike helm chart

Comment (akshaypatidar1999) [inline charts/kafka-connect/values.yaml:151]: This should be array. If we keep this as dict then we won't be able to use `valueFrom` in envs in deployment

File: charts/kafka-connect/values.yaml [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+## ref: https://kubernetes.io/docs/concepts/configuration/secret/
+secrets:
+  # username: kafka123
+  # password: connect321
+
+## @param livenessProbe the livenessProbe for the kafka connect pods
+## "livenessProbe" is required only for the edge cases where the custom script to be ran takes too much time
+## and errors by the ENTRYPOINT are ignored by the container
+## ref: https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/
+livenessProbe:
```
Comment (akshaypatidar1999) [inline charts/kafka-connect/values.yaml:256]: no readiness probe?

Comment (akshaypatidar1999) [inline charts/kafka-connect/values.yaml:256]: liveness/readiness probes should be enabled by default

File: charts/kafka-connect/README.md (hunk 2 above)
Comment (yash-saxena-d11) [reply charts/kafka-connect/README.md:50]: done

File: charts/kafka-connect/values.yaml (hunk 3 above)
Comment (yash-saxena-d11) [reply charts/kafka-connect/values.yaml:151]: done

Comment (yash-saxena-d11) [reply charts/kafka-connect/values.yaml:151]: done

File: charts/kafka-connect/values.yaml (hunk 4 above)
Comment (yash-saxena-d11) [reply charts/kafka-connect/values.yaml:256]: added

Comment (yash-saxena-d11) [reply charts/kafka-connect/values.yaml:256]: done

File: charts/kafka-connect/templates/NOTES.txt (hunk 1 above)
Comment (yash-saxena-d11) [reply charts/kafka-connect/templates/NOTES.txt:9]: done

File: charts/kafka-connect/values.yaml [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+  ##   clientIP:
+  ##     timeoutSeconds: 300
+  ##
+  sessionAffinityConfig: {}
+
+## The Kafka Connect properties
+## ref: https://docs.confluent.io/current/connect/userguide.html#configuring-workers
+configurationOverrides:
+  ##  @param configurationOverrides.plugin.path The plugin path in the file system
+  "plugin.path": "/usr/share/java,/usr/share/confluent-hub-components"
```
Comment (surajgour-d11) [inline charts/kafka-connect/values.yaml:126]: can we add "/usr/share/connect_plugins" here

File: charts/kafka-connect/ci/disable-probes.yaml [hunk 6]
```diff
@@ -0,0 +1,2 @@
+livenessProbe.enabled: false
+readinessProbe.enabled: false
```
Comment (akshaypatidar1999) [inline charts/kafka-connect/ci/disable-probes.yaml:2]: Add new line at end

File: charts/kafka-connect/templates/NOTES.txt [hunk 7]
```diff
@@ -0,0 +1,9 @@
+CHART NAME: {{ .Chart.Name }}
+CHART VERSION: {{ .Chart.Version }}
+APP VERSION: {{ .Chart.AppVersion }}
+
+** Please be patient while the chart is being deployed **
+
+Check the cluster status by running:
+
+{{ include "kafka-connect.fullname" . }}.{{ .Release.Namespace }}:{{ .Values.service.ports.kafkaConnect }}
```
Comment (akshaypatidar1999) [inline charts/kafka-connect/templates/NOTES.txt:9]: This won't be accessible from outside. Add a command to do `kubectl proxy` as well

Comment (yash-saxena-d11) [reply charts/kafka-connect/templates/NOTES.txt]: done

**Review by akshaypatidar1999 — APPROVED** [review]

**Pull request merged**